app/src/main/java/eu/zimbelstern/tournant/ui/SettingsActivity.kt

```diff
@@ -98,6 +100,30 @@ class SettingsActivity : AppCompatActivity() {
 				}
 			}
 
+			findPreference<SwitchPreference>("auto_backup")?.apply {
+				isEnabled = findPreference<ListPreference>("mode")?.value == "STANDALONE"
+				isChecked = sharedPrefs.getBoolean(PREF_AUTO_BACKUP, false)
+				setOnPreferenceChangeListener { _, value ->
+					sharedPrefs
+						.edit()
+						.putBoolean(PREF_AUTO_BACKUP, value as Boolean)
+						.apply()
+					if (value && sharedPrefs.getString(PREF_BACKUP_FILE, "").isNullOrEmpty()) {
+						(activity as SettingsActivity).chooseFile()
+					}
+					true
+				}
+			}
+
+			findPreference<Preference>("backup_file")?.apply {
+				isEnabled = findPreference<ListPreference>("mode")?.value == "STANDALONE"
+				summary = sharedPrefs.getString(PREF_BACKUP_FILE, "")
+				setOnPreferenceClickListener {
+					(activity as SettingsActivity).chooseFile()
```

Comment:
You are asking the user to choose a file here. But before the first backup, the user will not have a xml file to choose.